src/Symfony/Component/Messenger/Tests/Transport/Serialization/SerializerTest.php

```diff
@@ -61,4 +80,26 @@ public function testUsesTheCustomFormatAndContext()
         $this->assertSame('Yay', $encoded['body']);
         $this->assertSame($message, $decoded);
     }
+
+    public function testEncodedWithSerializationConfiguration()
+    {
+        $serializer = new Serializer(
+            new SerializerComponent\Serializer(array(new ObjectNormalizer()), array('json' => new JsonEncoder()))
+        );
+
+        $message = new ConfiguredMessage(new DummyMessage('Hello'));
+        $message
+            ->with(new SerializerConfiguration(array(ObjectNormalizer::GROUPS => array('foo'))))
+            ->with(new ValidationConfiguration(array('foo', 'bar')))
+        ;
+
+        $encoded = $serializer->encode($message);
+
+        $this->assertArrayHasKey('body', $encoded);
+        $this->assertArrayHasKey('headers', $encoded);
+        $this->assertArrayHasKey('type', $encoded['headers']);
+        $this->assertEquals(DummyMessage::class, $encoded['headers']['type']);
+        $this->assertArrayHasKey('X-Message-Configurations', $encoded['headers']);
+        $this->assertSame('a:2:{s:75:"Symfony\Component\Messenger\Transport\Serialization\SerializerConfiguration";C:75:"Symfony\Component\Messenger\Transport\Serialization\SerializerConfiguration":59:{a:1:{s:7:"context";a:1:{s:6:"groups";a:1:{i:0;s:3:"foo";}}}}s:76:"Symfony\Component\Messenger\Middleware\Configuration\ValidationConfiguration";C:76:"Symfony\Component\Messenger\Middleware\Configuration\ValidationConfiguration":79:{a:2:{s:6:"groups";a:2:{i:0;s:3:"foo";i:1;s:3:"bar";}s:14:"group_sequence";b:0;}}}', $encoded['headers']['X-Message-Configurations']);
```

Comment:
That's where a JSON encoding instead of the PHP `serialize` would be less verbose. But I'm not strongly opinionated on that (or even maybe omitting the keys in the array as we don't necessarily need them).

Reply:
Some wild thoughts: 
1. perhaps we shouldn't enforce `\Serializable` on configurations, but still implements it for core implems. That's still the preferred and pragmatic way to me considering the same stack must be used from end-to-end to handle configs. 
1. does every config need to survive transport? What about just dropping the one not serializable/implementing a specific interface?
1. we could still provide both json normalizers (so extra wiring for the `Messenger\Transport\Serialization\Serializer` to work properly) & `\Serializable` implems, so the messenger sf `Serializer` transport use it but serializable simple implems still exists if someone write its own transport not using the sf Serializer. My guess is that it'll probably by not very common not using the default serializer transport, though.